pallets/asset/src/lib.rs

```diff
@@ -2189,6 +2214,49 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
+    /// Accept and process a treasury transfer.
+    pub fn _accept_treasury_transfer(to_did: IdentityId, auth_id: u64) -> DispatchResult {
+        ensure!(
+            <identity::Authorizations<T>>::contains_key(Signatory::from(to_did), auth_id),
+            AuthorizationError::Invalid
+        );
+
+        let auth = <identity::Authorizations<T>>::get(Signatory::from(to_did), auth_id);
+
+        let (owner_did, ticker) = match auth.authorization_data {
+            AuthorizationData::TransferTreasury(ticker) => (auth.authorized_by, ticker),
+            _ => return Err(Error::<T>::NoTreasuryTransferAuth.into()),
+        };
+
+        let token = <Tokens<T>>::get(&ticker);
+
+        ensure!(
+            token.owner_did == owner_did,
+            Error::<T>::NotAnOwner
+        );
+
+        <identity::Module<T>>::consume_auth(
+            owner_did,
```

Comment:
You can directly pass `token.owner_did` to the `consume_auth` fn and it will automatically do the `token.owner_did == owner_did` check.